### PR TITLE
doc/design: Update the release documentation

### PR DESCRIPTION
**Pull request by timflannagan**
Update release.md now that the repository uses goreleaser to
automatically produce draft releases for new tags.

Signed-off-by: timflannagan <timflannagan@gmail.com>

**Description of the change:**

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: doc/design/release.md [hunk 1]
```diff
@@ -1,111 +1,45 @@
...
 * Pull the latest.
 * Make sure you are on `master` branch.
 * Make a new tag that matches the version.
 * Push tag directly to this repository.
 
 ```bash
-# 0.11.0 is the bumped version.
-git tag -a v0.18.0 -m "Version 0.18.0"
+# v0.20.0 is the bumped version.
+git tag -a v0.20.0 -m "Version 0.20.0"
```
Comment (joelanford) [inline doc/design/release.md:26]: Not necessarily related to the scope of this PR, but... should we sign our tags? (with the `--sign` flag)

Comment (timflannagan) [reply doc/design/release.md:26]: I don't have any opposition towards doing that. Might be worth bubbling that up to the rest of the team to see if there are any strong objections.

**Comment by timflannagan** [comment]
rebased against master to make branch protection happy.

File: doc/design/release.md (hunk 1 above)
Comment (awgreene) [reply doc/design/release.md:26]: I'm for it.

**Review by awgreene — APPROVED** [review]
/approve

Great work on this @timflannagan

**Comment by awgreene** [comment]
Ah - the PR had already been approved by @perdasilva, adding an lgtm.
/lgtm

**Comment by timflannagan** [comment]
Force merging as this PR only affects the release documentation and we don't need to continuously run the e2e-tests workflow to get a successful check for this be automatically merged.

**Pull request merged**